tests/api2/test_pool_dataset_track_processes.py

```diff
@@ -0,0 +1,24 @@
+import sys
+import os
+sys.path.append(os.getcwd())
+
+from functions import POST, make_ws_request
+from auto_config import pool_name, ip
+from middlewared.test.integration.utils import ssh
+
+TEST_DATASET = f'{pool_name}/testing_processes'
+
+
+def test_01_create_dataset():
+    results = POST('/pool/dataset', {'name': TEST_DATASET})
+    assert results.status_code == 200, results.text
+
+
+def test_02_open_path_and_check_proc():
+    test_file = f'/mnt/{TEST_DATASET}/test_file'
+    open_pid = ssh(f'python -c "import time; f = open(\"{test_file}\", \"w+\"); time.sleep(10)" & echo $!')
+    assert open_pid.isdigit()
+
+    payload = {'msg': 'method', 'method': 'pool.dataset.processes_using_paths', 'params': list(TEST_DATASET)}
+    res = make_ws_request(ip, payload)
+    assert int(res[0]['pid']) == int(open_pid)
```

Comment:
Great job, thank you for adding tests!

Can we please also:
* assert `name` and `cmdline` contents since the code under test is responsible for filling them
* assert that `len(res) == 1` to ensure that code under test does not add extra processes to the list
* kill that python process after we're done with the test
* destroy the test dataset after we're done with the test. The easiest way is to use `with dataset(...)` decorator like my other tests do and skip having fake tests like `test_01_create_dataset` completely. WIth that in mind we may stop adding numbers to the test names because they became independent units which execution order does not matter (like it should be ideally).

Also it might be nice to:
* sleep for some time between launching python process and asserting an open file because it might take quite a time for python interpreter to initialize if testing VM host is busy
* you might like how `from middlewared.test.integration.utils import call` works (just `call("method", arg1, arg2) `)